### PR TITLE
feat(cdn): add item gc

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14234.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by yesnault — APPROVED** [review]

File: engine/cdn/cdn_item.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		}
+	default:
+		return sdk.NewErrorFrom(sdk.ErrInvalidData, "unable to complete item: unknown item type %s", item.Type)
+	}
+
+	h, err := convergent.NewHash(reader)
+	if err != nil {
+		return err
+	}
+	item.Hash = hex.EncodeToString(h.Sum(nil))
```
Comment (fsamin) [inline engine/cdn/cdn_item.go:51]: with the symmecrypt update, convergent.NewHash returns a string, so you don't need to do `hex.EncodeToString(h.Sum(nil))`

File: engine/cdn/index/dao.go [hunk 2]
```diff
@@ -77,3 +92,14 @@ func LoadItemByApiRefHashAndType(ctx context.Context, m *gorpmapper.Mapper, db g
...
 	return getItem(ctx, m, db, query)
 }
+
+func LoadOldItemIDsByStatusAndDuration(db gorp.SqlExecutor, status string, duration int) ([]string, error) {
+	query := gorpmapper.NewQuery(`
+		SELECT id
+		FROM index
+		WHERE
+			status = $1 AND
+            last_modified < NOW() - $2 * INTERVAL '1 second'
```
Comment (fsamin) [inline engine/cdn/index/dao.go:102]: order by ?

File: engine/cdn/index/dao.go [hunk 3]
```diff
@@ -77,3 +92,14 @@ func LoadItemByApiRefHashAndType(ctx context.Context, m *gorpmapper.Mapper, db g
 	`).Args(hash, typ)
 	return getItem(ctx, m, db, query)
 }
+
+func LoadOldItemIDsByStatusAndDuration(db gorp.SqlExecutor, status string, duration int) ([]string, error) {
```
Comment (fsamin) [inline engine/cdn/index/dao.go:96]: add options if you need decryption

**Review by fsamin — CHANGES_REQUESTED** [review]

**Review by richardlt — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14261.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14261.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14269.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by yesnault — APPROVED** [review]

**Pull request merged**